gui/app/batteryinfo.h

```diff
@@ -0,0 +1,25 @@
+#ifndef BATTERYINFO_H
+#define BATTERYINFO_H
+
+#include <QObject>
+
+/**
+ * @brief The BatteryInfo class
+ *
+ * Keeps remaining percetange of battery time.
+ */
+class BatteryInfo : public QObject {
```

Comment:
I think this class should be merged into GuiStateContainer, which by design contains all the GUI state.

Reply:
I added as a property of GuiStateContainer

Reply:
Why not just have battery percentage as a property of GuiStateContainer? As-is, we have two levels of "changed" signals - "battery info changed" (which can never happen) and "percentage changed within the current battery info".